### DISCUSSION TITLE
Questions about time calculation

### DISCUSSION THREAD

**Discussion by tanzhengo** [larics/icuas24_competition#20] [General]
1. whether the final time is calculated after starting the simulation or after the drone takes off？

2. I would like to ask if the front and back sides of the same fruit are counted once at the same position, and be accumulated at different positions?

**Comment by fpetric** [comment]
> whether the final time is calculated after starting the simulation or after the drone takes off？

The time starts when the message is published on the `challenge_started` topic

> I would like to ask if the front and back sides of the same fruit are counted once at the same position, and be accumulated at different positions?

If the fruits are at the same position, they should be counted only once even when visible from both sides.

**Reply by cmakelabs** [reply]
Just to clarify, I think there is still confusion, so the drone should any ways visit both sides and check if the fruits are in the same position, then ignore it, if not then add them to the count of the first side?
Like if I have 2 tomatoes on the first side, and on the second side I have two tomatoes on the same position as the ones were on the first side then I do not count them, and my total count will still be 2,
but if the position is different then I include them in the count, and the total count will be 4? Is that exactly what you meant?

**Reply by fpetric** [reply]
Yes. Note here that I am talking about the "same" 3D position. 
![same_fruits](https://github.com/larics/icuas24_competition/assets/5979567/13dde6a3-e117-4cc2-a64b-e16761628390)
These would be examples of the fruit at the same position.

**Reply by cmakelabs** [reply]
so in this case, we count 2 from the front and 1 from the back? with a total of 3?

**Reply by fpetric** [reply]
Yes. You can check .dae files in the models folder to verify count for each plant

**Comment by tanzhengo** [comment]
> Yes. Note here that I am talking about the "same" 3D position. ![same_fruits](https://private-user-images.githubusercontent.com/5979567/295261828-13dde6a3-e117-4cc2-a64b-e16761628390.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4NzAxNzgsIm5iZiI6MTcwNDg2OTg3OCwicGF0aCI6Ii81OTc5NTY3LzI5NTI2MTgyOC0xM2RkZTZhMy1lMTE3LTRjYzItYTY0Yi1lMTY3NjE2MjgzOTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTBUMDY1NzU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDVhYzhkY2NlMjIzNTRmYWI0NWFlYmEyZjY0NmZlMWI4YzFjMWRlY2RiYTcyZWQ2OWEyMGQwNTdmMGEwMWIyNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0gIN9rA-h1KfFO2D9Z18dwZ_4cACIEECEZ0Ed35njTw) These would be examples of the fruit at the same position.

May I ask if the brightness is different from the front and back sides? OR is it a display issue with the images?

**Reply by fpetric** [reply]
I think this is just due to Blender default light. The brightness should be the same from both sides in Gazebo.
